notebooks/predicting_queries/simple_intent_suggestion.ipynb

```diff
@@ -0,0 +1,570 @@
+   "outputs": [],
+   "source": [
+    "# Taxons for every bit of content \n",
+    "#taxons_df =graph.run(\"MATCH (c:Cid)-[r:IS_TAGGED_TO]->(t:Taxon)RETURN c.name as pagePath,t.name as taxon\").to_data_frame()\n",
+    "#taxons_df.to_csv(PROCESSED_DATA+'/taxons.csv')\n",
+    "taxons_df =pd.read_csv(PROJECT_DIR+'/data/processed/taxons.csv')\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": []
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# inner join taxons to session/query data \n",
+    "# We'll end up with multiple taxons per query \n",
+    "# ie each query has potentially many pages viewed prior to the query being made AND pages can have multiple taxons\n",
+    "\n",
+    "query_taxons = queries_df.merge(taxons_df,on='pagePath')\n",
+    "# Split queries up (they're stupidily aggregated in the sql)\n",
+    "query_taxons['query']=query_taxons['search_terms'].map(lambda x: x.split(','))\n",
+    "query_taxons = query_taxons.explode('query')\n",
+    "# Average number of taxons visited per query (around 8)\n",
+    "np.mean(query_taxons.groupby('session_id').size())"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Text preprocessing\n",
+    "translator = str.maketrans('', '', string.punctuation)\n",
+    "# Some multiword tokens\n",
+    "tokenizer = MWETokenizer([('log', 'in'), ('sign', 'in'), ('sign', 'up')])\n",
+    "# Tokenise query, explode, and regroup by taxon \n",
+    "# End up with a list of tokens per taxon\n",
+    "query_taxons['tokens'] = query_taxons['query'].map(lambda x: tokenizer.tokenize(x.translate(translator).split()))\n",
+    "query_taxons = query_taxons.explode('tokens')\n",
+    "query_taxons = query_taxons[['taxon','tokens']]\n",
+    "query_taxons = query_taxons.dropna()\n",
+    "# Group query tokens by taxon so we have a list of tokens per taxon\n",
+    "query_taxons = query_taxons.groupby('taxon').aggregate(lambda x: list(x)).reset_index()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# All the maintstream content text,titles from gov uk (assumed targets for users)\n",
+    "#title_df =graph.run(\"match (m:Mainstream) return m.name as name, m.title as title, m.text as text\").to_data_frame()\n",
+    "#title_df.to_csv(PROCESSED_DATA+'/mainstreamcontent.csv')\n",
+    "title_df = pd.read_csv(PROJECT_DIR+'/data/processed/mainstreamcontent.csv')\n",
+    "title_df = title_df.dropna()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#We want to weight titles a bit highly, so err just multiply by 3\n",
+    "title_df['body_tokens'] = title_df['text'] + ' ' + (title_df['title'] + ' ' * 3)\n",
```

Comment:
How is this weighting it by three, exactly? If I run the following two lines of code below, they look the same:
```python
title_df['title']
title_df['title'] + ' ' * 3
```
However, the below looks more like what you're trying to do and is what's in the `simple_entity_suggestion.ipynb` notebook.